src/irmin-pack/atomic_write_intf.ml

```diff
@@ -17,7 +17,7 @@
 module type S = sig
   include Irmin.Atomic_write.S
 
-  val flush : t -> unit
+  val sync : t -> unit
```

Comment:
I think it's better to keep it calling `flush` for write instances.

Reply:
I've remove `flush` and added `sync`.

Flush was a no-op because `set_entry` always flushes at the end. https://github.com/mirage/irmin/blob/main/src/irmin-pack/atomic_write.ml#L82

Sync was automatically called before each `unsafe_find`, which is unsafe because it may catch branches that point to a non-flushed index

Reply:
Agree with `flush`, good catch.

For the `sync` do you mean that the index/pack.store is maybe not flushed when the branch store is? I'm wondering if the problem isn't that `set` does not force the pack file to flush (which will force the index).  
`set` calls `W.notify`. I'm not sure how watches are used, but don't we need to guarantee that both the branch tag and the commit reach the disk? (cc @zshipko ).

Reply:
Right now `irmin-pack` watches don't listen for changes on disk (which is something I'd like to improve at some point). Watchers are directly notified when the process they created the watch in does a set/remove operation.

Reply:
In general, we use some words in unusual ways. I would say that sync is one such word. It tends to be linked in the mind of a programmer to `fsync`. However, in our codebase, sync tends to be a special function that is used by RO instances to detect changes in underlying files, and somehow force a reload of in-memory structures. So, the flow of information is somehow the opposite of the normal `fsync`. Whatever, I think we need to clearly document what "sync" is, what it is supposed to do etc. If it is for RO instances, we should make this clear. Then we can decide whether the implementation actually does what it is supposed to do.

Reply:
Shoud we use push/pull instead?

Reply:
I find `push`, `pull` and `fetch` very connotated to interactions with a remote server.  

`flush`/`reload`?